src/Castle.Core.Tests/DynamicProxy.Tests/InvocationProceedInfoTestCase.cs

```diff
@@ -0,0 +1,97 @@
+// Copyright 2004-2019 Castle Project - http://www.castleproject.org/
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+namespace Castle.DynamicProxy.Tests
+{
+	using System;
+
+	using Castle.DynamicProxy.Tests.Interceptors;
+	using Castle.DynamicProxy.Tests.InterClasses;
+	using Castle.DynamicProxy.Tests.Interfaces;
+
+	using NUnit.Framework;
+
+	[TestFixture]
+	public class InvocationProceedInfoTestCase
+	{
+		private readonly ProxyGenerator generator = new ProxyGenerator();
+
+		[Test]
+		public void Proxy_without_target_and_last_interceptor_ProceedInfo_succeeds()
+		{
+			var interceptors = new IInterceptor[]
+			{
+				new SetReturnValueInterceptor(0),
```

Comment:
Does this and the next unit test do what you expect, `SetReturnValueInterceptor` doesn't proceed so the `WithCallbackInterceptor` wouldn't get called. Maybe I'm just misreading it.